core/project_settings.cpp

```diff
@@ -79,6 +79,12 @@ String ProjectSettings::localize_path(const String &p_path) const {
 			return p_path;
 		};
 
+		Vector<String> res_arr = resource_path.split("/");
+		Vector<String> cwd_arr = cwd.split("/");
+		if (res_arr[res_arr.size() - 1] != cwd_arr[res_arr.size() - 1]) {
```

Comment:
I don't really understand what this check does. A comment would be good to explain why this condition means that we should return the unchanged `p_path`.

Reply:
@akien-mga added comment